pallets/pallet-bonded-coins/src/benchmarking.rs

```diff
@@ -1 +1,855 @@
+{
+	fn calculate_collateral_asset_id(seed: u32) -> CollateralAssetIdOf<T> {
+		seed.into()
+	}
+
+	fn calculate_bonded_asset_id(seed: u32) -> FungiblesAssetIdOf<T> {
+		seed.into()
+	}
+}
+
+#[benchmarks(where
+	<CurveParameterTypeOf<T> as Fixed>::Bits: Copy + ToFixed + AddAssign + BitOrAssign + ShlAssign + TryFrom<U256>,
+	CollateralCurrenciesBalanceOf<T>: Into<U256> + TryFrom<U256>,
+	<T as Config>::CollateralCurrencies: Create<T::AccountId> ,
+	<T as Config>::Fungibles: InspectRoles<T::AccountId> + AccountTouch<FungiblesAssetIdOf<T>, AccountIdOf<T>>,
+	<T as Config>::DepositCurrency: Mutate<T::AccountId>,
+	<T as Config>::CollateralCurrencies: MutateFungibles<T::AccountId>,
+	AccountIdLookupOf<T>: From<T::AccountId>,
+)]
+mod benchmarks {
+	use frame_support::traits::OriginTrait;
+	use frame_support::traits::{
+		fungible::{Inspect, Mutate, MutateHold},
+		fungibles::{Destroy, Inspect as InspectFungibles, Mutate as MutateFungibles},
+	};
+	use frame_support::traits::{fungibles::Create, AccountTouch, EnsureOrigin, Get};
+	use sp_runtime::{traits::Zero, BoundedVec, SaturatedConversion};
+	use sp_std::ops::Mul;
+
+	use crate::{
+		curves::Curve,
+		mock::*,
+		types::{Locks, PoolManagingTeam, PoolStatus},
+		AccountIdLookupOf, AccountIdOf, CollateralAssetIdOf, CurveParameterInputOf, HoldReason, PoolDetailsOf, Pools,
+		TokenMetaOf,
+	};
+
+	use super::*;
+
+	// helper functions
+	// collateral currencies
+	fn create_collateral_asset<T: Config>(asset_id: CollateralAssetIdOf<T>)
+	where
+		<T as Config>::CollateralCurrencies: Create<T::AccountId>,
+	{
+		let pool_account = account("collateral_owner", 0, 0);
+		<T as Config>::CollateralCurrencies::create(asset_id, pool_account, false, 1u128.saturated_into())
+			.expect("Creating collateral asset should work");
+	}
+
+	fn calculate_default_collateral_asset_id<T: Config>() -> CollateralAssetIdOf<T> {
+		T::BenchmarkHelper::calculate_collateral_asset_id(u32::MAX)
+	}
+
+	fn create_default_collateral_asset<T: Config>() -> CollateralAssetIdOf<T>
+	where
+		<T as Config>::CollateralCurrencies: Create<T::AccountId>,
+	{
+		let collateral_id = calculate_default_collateral_asset_id::<T>();
+		create_collateral_asset::<T>(collateral_id.clone());
+		collateral_id
+	}
+
+	fn set_collateral_balance<T: Config>(asset_id: CollateralAssetIdOf<T>, who: &AccountIdOf<T>, amount: u128)
+	where
+		<T as Config>::CollateralCurrencies: MutateFungibles<T::AccountId>,
+	{
+		T::CollateralCurrencies::set_balance(asset_id, who, amount.saturated_into());
+	}
+
+	// bonded currencies
+
+	fn create_bonded_asset<T: Config>(asset_id: T::AssetId) {
+		let pool_account = account("bonded_owner", 0, 0);
+		<T as Config>::Fungibles::create(asset_id, pool_account, false, 1u128.saturated_into())
+			.expect("Creating bonded asset should work");
+	}
+
+	fn create_bonded_currencies_in_range<T: Config>(c: u32, is_destroying: bool) -> Vec<FungiblesAssetIdOf<T>> {
+		let mut asset_ids = Vec::new();
+		for i in 0..c {
```

Comment:
I know it's very nit-picky, but I think it would be slitghly more readable to replace this with `1..=c`, to use the same logic as the linear parameter computation.

Reply:
[470cad9](https://github.com/KILTprotocol/kilt-node/pull/789/commits/470cad95496101a3eb7873205c3196ebffd738fb)